auditwheel/policy/musllinux-policy.json

```
@@ -0,0 +1,48 @@
+[
+  {"name": "linux",
+   "aliases": [],
+   "priority": 0,
+   "symbol_versions": {},
+   "lib_whitelist": []
+  },
+  {"name": "musllinux_1_1",
+    "aliases": [],
+    "priority": 100,
+    "symbol_versions": {
+      "i686": {
+      },
+      "x86_64": {
+      },
+      "aarch64": {
+      },
+      "ppc64le": {
+      },
+      "s390x": {
+      },
+      "armv7l": {
+      }
+    },
+    "lib_whitelist": [
+      "libc.so"
+    ]},
+    {"name": "musllinux_1_2",
+    "aliases": [],
+    "priority": 90,
+    "symbol_versions": {
+      "i686": {
+      },
+      "x86_64": {
+      },
+      "aarch64": {
+      },
+      "ppc64le": {
+      },
+      "s390x": {
+      },
+      "armv7l": {
+      }
```

```suggestion
      }
      "armv6l": {
      }
```

It's still happening that a small system with musl also runs armv6. Unlike manylinux, musllinux references the system arch on the PEP and does not explicitly take armv6 out.

---

> Unlike manylinux, musllinux references the system arch on the PEP and does not explicitly take armv6 out.

[PEP 600 does not rule armv6 either](https://www.python.org/dev/peps/pep-0600/#core-definition) however, it requires  a new PEP for armv6 since it has multiple variants (at least soft-float & hard-float ABIs are both returned as `armv6l`). This was similar for armv7l but, given [PEP 599](https://www.python.org/dev/peps/pep-0599/), it represents the hard-float ABI when used for manylinux (& now musllinux)...

[PEP 656 specification](https://www.python.org/dev/peps/pep-0656/#specification) refers to the PEP 600 one where there's an example that I'll reproduce below:

> Example: The imaginary architecture LEG v7 has both big-endian and little-endian variants. Big-endian binaries require a big-endian system, and little-endian binaries require a little-endian system. But unfortunately, it's discovered that due to a bug in PEP 425, both variants use the same architecture tag, legv7. This makes it impossible to create a compliant manylinux_2_17_legv7 wheel: no matter what we do, it will crash on some user's systems. So, we write a new PEP defining architecture tags legv7le and legv7be; now we can ship manylinux LEG v7 wheels.

I wish a PEP had been the case for `armv7l` where `armhf` (as used by many distros) or `armv7hf` would have been clearer than `armv7l`.

---

I think it's many what the Linux kernel will return, and that can be only armv7l and armv6l. Anyway, I'm fine with the answer; I will patch it locally for Home Assistant to build musl wheel for our usage.

---

> I think it's what the Linux kernel will return, and that can be only armv7l and armv6l. 

You're right and that's because of this shortcoming of `distutils.util.get_platform()` that there are such patch in wheel & packaging:

https://github.com/pypa/wheel/blob/0b29b695ae458450217552978ad3094ed06a16d7/src/wheel/bdist_wheel.py#L47-L56

https://github.com/pypa/packaging/blob/e2bc8f6ef2d66f6eaaf522fcf70a0f203b734064/packaging/tags.py#L417-L427
https://github.com/pypa/packaging/blob/e2bc8f6ef2d66f6eaaf522fcf70a0f203b734064/packaging/_manylinux.py#L114-L119